core/server/common/src/main/java/alluxio/master/journal/JournalUtils.java

```diff
@@ -103,7 +104,16 @@ public static void restoreJournalEntryCheckpoint(CheckpointInputStream input, Jo
     JournalEntryStreamReader reader = new JournalEntryStreamReader(input);
     JournalEntry entry;
     while ((entry = reader.readEntry()) != null) {
-      journaled.processJournalEntry(entry);
+      try {
+        journaled.processJournalEntry(entry);
+      } catch (Throwable t) {
+        if (ServerConfiguration.getBoolean(PropertyKey.MASTER_JOURNAL_TOLERATE_CORRUPTION)) {
+          ProcessUtils.fatalError(true,  LOG, t,
+              "Failed to process journal entry %s from a journal checkpoint", entry);
+        } else {
+          throw t;
```

Comment:
I think we want to replay the rest of the journal still, not throw an exception

Reply:
If not tolerate corruption, we directly throw any throwables.
If tolerate corruption, not exception will be thrown and we will still process the other journal entries.